generator.yaml

```diff
@@ -17,12 +18,131 @@ ignore:
   shape_names: null
 model_name: route53
 operations:
+  ChangeResourceRecordSets:
+    operation_type:
+    - Create
+    - Delete
+    resource_name:
+      RecordSet
+  ListResourceRecordSets:
+    operation_type:
+    - List
+    resource_name:
+      RecordSet
   CreateHostedZone:
     output_wrapper_field_path: HostedZone
   GetHostedZone:
     output_wrapper_field_path: HostedZone
 prefix_config: {}
 resources:
+  RecordSet:
+    exceptions:
+      terminal_codes:
+      - InvalidChangeBatch
+      - InvalidInput
+      - NoSuchHostedZone
+      - NoSuchHealthCheck
+    fields:
+      AliasTarget:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.AliasTarget
+      # This field cannot be ignored, but values for this will be overriden and should not be used
+      ChangeBatch:
+        is_required: false
+      CidrRoutingConfig:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.CidrRoutingConfig
+      Failover:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.Failover
+      GeoLocation:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.GeoLocation
+      HealthCheckId:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.HealthCheckId
+      # Changing this value after a CR has been created could result in orphaned record sets
+      HostedZoneId:
+        references:
+          resource: HostedZone
+          path: Status.ID
+        is_required: true
+        is_immutable: true
+      ID:
+        is_primary_key: true
+        documentation: "ChangeID is used as a unique identifier"
```

Comment:
It seems that RecordSets don't really have a single field that represents a unique identifier. Instead, multiple fields are used to guarantee the uniqueness of a record set:
- DNSName
- HostedZoneID
- RecordType
- SetIdentifier 

Here, I've set the ID as the primary key for now. This is a `ChangeID` that represents changes made when `ChangeResourceRecordSets` is invoked. This ID is used in our reconcile logic to get the current status of a particular change (i.e. have the changes been synced to all DNS servers).